main/lsp/LSPIndexer.cc

```diff
@@ -259,14 +249,18 @@ LSPFileUpdates LSPIndexer::commitEdit(SorbetWorkspaceEditParams &edit) {
         // which one it will be.
         initialGS->errorQueue = make_shared<core::ErrorQueue>(
             initialGS->errorQueue->logger, initialGS->errorQueue->tracer, make_shared<core::NullFlusher>());
-        auto trees = pipeline::index(initialGS, frefs, config->opts, *emptyWorkers, kvstore);
+        auto trees = hashing::Hashing::indexAndComputeFileHashes(initialGS, config->opts, *config->logger, frefs,
+                                                                 *emptyWorkers, kvstore);
         update.updatedFileIndexes.resize(trees.size());
         for (auto &ast : trees) {
             const int i = fileToPos[ast.file];
             update.updatedFileIndexes[i] = move(ast);
         }
     }
 
+    // _Now_ that we've computed file hashes, we can make a fast path determination.
+    update.canTakeFastPath = canTakeFastPath(update, newlyEvictedFiles);
```

Comment:
We used to run `canTakeFastPath` and _then_ update `GlobalState` with the new files.

Now, we update `GlobalState` and _then_ run `canTakeFastPath`. Thus, we need to pass in the old files, pre-update, which contain their old pre-update hashes.